src/Aspire.Hosting.Azure.AppService/AzureAppServiceEnvironmentExtensions.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Aspire.Hosting.Azure.AppService;
+using Aspire.Hosting.Lifecycle;
+using Azure.Provisioning;
+using Azure.Provisioning.AppService;
+using Azure.Provisioning.ContainerRegistry;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Roles;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Extensions for adding Azure App Service Environment resources to a distributed application builder.
+/// </summary>
+public static partial class AzureAppServiceEnvironmentExtensions
+{
+    /// <summary>
+    /// Adds a azure app service environment resource to the distributed application builder.
+    /// </summary>
+    /// <param name="builder">The distributed application builder.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <returns><see cref="IResourceBuilder{T}"/></returns>
+    public static IResourceBuilder<AzureAppServiceEnvironmentResource> AddAppServiceEnvironment(this IDistributedApplicationBuilder builder, string name)
```

Comment:
`AddAzureAppServiceEnvironment`

Reply:
Resource types end with the *Resoure suffix.